bridge_adaptivity/module/views.py

```diff
@@ -73,31 +72,55 @@ def form_valid(self, form):
 @method_decorator(login_required, name='dispatch')
 class CourseDelete(BaseCourseView, GroupEditFormMixin, DeleteView):
     def get_success_url(self):
-        return reverse('module:course-list')
+        return self.request.GET.get('return_url') or reverse('module:course-list')
 
     def get(self, *args, **kwargs):
         return self.post(*args, **kwargs)
 
 
 @method_decorator(login_required, name='dispatch')
-class CourseAddGroup(FormView):
+class CourseAddGroup(JsonResponseMixin, FormView):
     model = CollectionGroup
     template_name = 'module/modals/course_add_group.html'
     form_class = AddCourseGroupForm
 
+    def get_success_url(self):
+        return self.request.GET.get('return_url') or reverse('module:course-detail', kwargs=self.kwargs)
+
     def get_form_kwargs(self):
         kwargs = super(CourseAddGroup, self).get_form_kwargs()
         kwargs['user'] = self.request.user
+        kwargs['course'] = get_object_or_404(Course, slug=self.kwargs['course_slug'])
         return kwargs
 
-    def form_valid(self, form):
+
+@method_decorator(login_required, name='dispatch')
+class CourseRmGroup(UpdateView):
```

Comment:
This view doesn't work. With the last updates

Reply:
fixed